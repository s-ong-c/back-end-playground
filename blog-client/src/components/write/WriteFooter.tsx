import * as React from 'react';
import styled from 'styled-components';
import Button from '../common/Button';

const WriteFooterBlock = styled.div`
  padding-left: 1rem;
  padding-right: 1rem;
  height: 4rem;
  width: 100%;
  box-shadow: 0px 0px 8px rgba(0, 0, 0, 0.1);
  background: rgba(255, 255, 255, 0.85);
  left: 0;
  top: 0;
  display: flex;
  justify-content: flex-end;
  align-items: center;
`;

const StyledButton = styled(Button)`
  height: 2.5rem;
  font-size: 1.125rem;
  & + & {
    margin-left: 0.75rem;
  }
`;
interface WriteFooterProps {}

const WriteFooter: React.SFC<WriteFooterProps> = props => {
  return (
    <WriteFooterBlock>
      <StyledButton color="lightGray" inline>
        임시저장
      </StyledButton>
      <StyledButton inline>출간하기</StyledButton>
    </WriteFooterBlock>
  );
};

export default WriteFooter;
