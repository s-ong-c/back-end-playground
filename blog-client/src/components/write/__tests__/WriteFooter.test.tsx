import * as React from 'react';
import { render, fireEvent } from 'react-testing-library';
import WriteFooter, { WriteFooterProps } from '../WriteFooter';

describe('WriteFooter', () => {
  const setup = (props: Partial<WriteFooterProps> = {}) => {
    const initialProps: WriteFooterProps = {
      onTempSave: () => {},
      onPublish: () => {},
    };
    const utils = render(<WriteFooter {...initialProps} {...props} />);

    const buttons = {
      tempSave: utils.getByText('임시저장'),
      publish: utils.getByText('출간하기'),
    };
    return {
      ...utils,
      buttons,
    };
  };
  it('renders property', () => {
    setup();
  });
  it('matches snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
  it('buttons are working property', () => {
    const onPublish = jest.fn();
    const onTempSave = jest.fn();
    const utils = setup({ onPublish, onTempSave });
    const { tempSave, publish } = utils.buttons;
    fireEvent.click(tempSave);
    expect(onTempSave).toBeCalled();
    fireEvent.click(publish);
    expect(onPublish).toBeCalled();
  });
});
