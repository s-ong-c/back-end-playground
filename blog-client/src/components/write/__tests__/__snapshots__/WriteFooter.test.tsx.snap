// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WriteFooter matches snapshot 1`] = `
<div>
  <div
    class="sc-bwzfXH itiXzj"
  >
    <button
      class="sc-htpNat bGXGEN sc-bdVaJa hOxnkh"
      color="lightGray"
    >
      임시저장
    </button>
    <button
      class="sc-htpNat bGXGEN sc-bdVaJa hWgIMi"
      color="teal"
    >
      출간하기
    </button>
  </div>
</div>
`;
